httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -233,6 +231,8 @@ public Reader asReader(Charset charset) throws IOException {
       @Override
       public void close() throws IOException {
         EntityUtils.consume(entity);
+        if (httpResponse instanceof CloseableHttpResponse)
```

Comment:
```suggestion
        try {
          EntityUtils.consume(entity);
        } finally {
          if (httpResponse instanceof CloseableHttpResponse)
            ((CloseableHttpResponse) httpResponse).close();
        }
```

Reply:
hi @velo 
Please do have a look at this, when you are free.
I do not have write permissions to apply this change.
Thank you